frontend/src/components/Board/BuyParts/index.tsx

```diff
@@ -0,0 +1,464 @@
+  const bom: Array<farnellRow> = [['Part Number', 'Quantity', 'Line Note']]
+  for (const line of lines) {
+    if (line.retailers.Farnell || line.retailers.Newark) {
+      const row: farnellRow = [
+        line.retailers.Farnell || line.retailers.Newark,
+        calculateQuantity(line.quantity, multiplier, addPercent).toString(),
+        line.description,
+      ]
+      bom.push(row)
+    }
+  }
+  return bom
+}
+
+const mouserBom = (lines: Array<Line>, multiplier: number, addPercent: number) => {
+  const bom: Array<mouserRow> = [
+    [
+      'Mouser Part Number',
+      'Mfr Part Number',
+      'Mfr Name',
+      'Description',
+      'Quantity',
+      'Customer Part Number',
+    ],
+  ]
+
+  for (const line of lines) {
+    if (line.retailers?.Mouser) {
+      const row: mouserRow = [
+        line.retailers.Mouser,
+        line.partNumbers?.[0]?.manufacturer,
+        line.partNumbers?.[0]?.part,
+        line.description,
+        calculateQuantity(line.quantity, multiplier, addPercent).toString(),
+        line.row,
+      ]
+      bom.push(row)
+    }
+  }
+
+  return bom
+}
+
+const lcscBom = (lines: Array<Line>, multiplier: number, addPercent: number) => {
+  const bom: Array<lcscRow> = [
+    [
+      'Quantity',
+      'LCSC Part Number',
+      'Manufacturer',
+      'Manufacturer Part Number',
+      'Description',
+      'Customer Part Number',
+    ],
+  ]
+
+  for (const line of lines) {
+    const row: lcscRow = [
+      calculateQuantity(line.quantity, multiplier, addPercent).toString(),
+      line.retailers?.LCSC,
+      line.partNumbers?.[0]?.manufacturer,
+      line.partNumbers?.[0]?.part,
+      line.description,
+      line.row,
+    ]
+    bom.push(row)
+  }
+
+  return bom
+}
+type Retailer = 'RS' | 'Newark' | 'Farnell' | 'Mouser' | 'LCSC' | 'Digikey'
+
+const csvBom = (
+  lines: Array<Line>,
+  multiplier: number,
+  addPercent: number,
+  retailer: Retailer,
+) => {
+  switch (retailer) {
+    case 'RS':
+      return rsBom(lines, multiplier, addPercent)
+    // Newark and Farnell have the same file format.
+    case 'Newark':
+    case 'Farnell':
+      return farnellBom(lines, multiplier, addPercent)
+    case 'Mouser':
+      return mouserBom(lines, multiplier, addPercent)
+    case 'LCSC':
+      return lcscBom(lines, multiplier, addPercent)
+    default:
+      throw new Error(`Unknown retailer: ${retailer}`)
+  }
+}
+
+export const calculateQuantity = (
+  quantity: number,
+  multiplier: number,
+  addPercent: number,
+) => {
+  let newQuantity = Math.ceil(quantity * multiplier)
+  if (addPercent > 0) {
+    newQuantity += Math.ceil(newQuantity * (addPercent / 100))
+  }
+  return newQuantity
+}
+
+function downloadBomCSV({
+  name,
+  lines,
+  multiplier,
+  buyAddPercent,
+  retailer,
+}: Order) {
+  const csvContent = `${csvBom(lines, multiplier, buyAddPercent, retailer)
+    .map((e: Array<string>) => e.join(','))
+    .join('\n')}\n`
```

Comment:
- Strings with commas in them need to be quoted
- Strings with quotes in them need to be escaped